pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -223,15 +223,16 @@ func (b *OFBridge) createGroupWithType(id GroupIDType, groupType ofctrl.GroupTyp
 	return g
 }
 
-func (b *OFBridge) DeleteGroup(id GroupIDType) bool {
-	g := b.ofSwitch.GetGroup(uint32(id))
-	if g == nil {
-		return true
+func (b *OFBridge) DeleteGroup(id GroupIDType) error {
+	ofctrlGroup := b.ofSwitch.GetGroup(uint32(id))
+	if ofctrlGroup == nil {
+		return nil
 	}
+	g := &ofGroup{bridge: b, ofctrl: ofctrlGroup}
 	if err := g.Delete(); err != nil {
-		return false
+		return fmt.Errorf("failed to delete the group in bundle: %w", err)
 	}
-	return true
+	return b.ofSwitch.DeleteGroup(uint32(id))
```

Comment:
Why needs this `b.ofSwitch.DeleteGroup(uint32(id))`? 

g.Delete() should already delete the group on OVS, and `ofGroup.Add()` does not add any object into ofctrl cache.

Reply:
We need `b.ofSwitch.DeleteGroup(uint32(id))` to delete the group id cache in `OFSwitch`. We can see that when creating a new group:

```go
// Create a new group. return an error if it already exists
func (self *OFSwitch) NewGroup(groupId uint32, groupType GroupType) (*Group, error) {
	// check if the group already exists
	if self.groupDb[groupId] != nil {
		return nil, errors.New("group already exists")
	}

	// Create a new group
	group := newGroup(groupId, groupType, self)
	// Save it in the DB
	self.groupDb[groupId] = group

	return group, nil
}
```

`g.Add()` and `g.Delete()` don't change the group id cache in `OFSwitch` since they are sent to OVS by bundle:

```go
func (g *ofGroup) Add() error {
	return g.bridge.AddOFEntriesInBundle([]OFEntry{g}, nil, nil)
}

func (g *ofGroup) Modify() error {
	return g.bridge.AddOFEntriesInBundle(nil, []OFEntry{g}, nil)
}

func (g *ofGroup) Delete() error {
	return g.bridge.AddOFEntriesInBundle(nil, nil, []OFEntry{g})
}
```
As a result, when deleting a group, we need also to delete the group id from `OFSwitch` cache.